player/src/androidTest/kotlin/com/tidal/sdk/player/playlog/PlayLogTest.kt

```diff
@@ -203,6 +208,87 @@ class PlayLogTest {
         )
     }
 
+    @Test
+    fun loadAndPlayThenPauseThenPlayNoNulls() =
+        loadAndPlayThenPauseThenPlay(MediaProduct(ProductType.TRACK, "1", "TESTA", "456"))
+
+    @Test
+    fun loadAndPlayThenPauseThenPlayNullSourceType() =
+        loadAndPlayThenPauseThenPlay(MediaProduct(ProductType.TRACK, "1", null, "789"))
+
+    @Test
+    fun loadAndPlayThenPauseThenPlayNullSourceId() =
+        loadAndPlayThenPauseThenPlay(MediaProduct(ProductType.TRACK, "1", "TESTB", null))
+
+    @Test
+    fun loadAndPlayThenPauseThenPlayNullSourceTypeNullSourceId() =
+        loadAndPlayThenPauseThenPlay(MediaProduct(ProductType.TRACK, "1", null, null))
+
+    @Suppress("LongMethod")
+    private fun loadAndPlayThenPauseThenPlay(mediaProduct: MediaProduct) = runTest {
+        val gson = Gson()
+        responseDispatcher[
+            "https://api.tidal.com/v1/tracks/1/playbackinfo?playbackmode=STREAM&assetpresentation=FULL&audioquality=LOW".toHttpUrl(),
+        ] = {
+            MockResponse().setBodyFromFile(
+                "api-responses/playbackinfo/tracks/playlogtest/get_1_bts.json",
+            )
+        }
+        responseDispatcher["https://test.audio.tidal.com/1_bts.m4a".toHttpUrl()] = {
+            MockResponse().setBodyFromFile("raw/playlogtest/1_bts.m4a")
+        }
+
+        player.playbackEngine.load(mediaProduct)
+        player.playbackEngine.play()
+        withContext(Dispatchers.Default.limitedParallelism(1)) {
+            withTimeout(4.seconds) {
+                player.playbackEngine.events.filter { it is Event.MediaProductTransition }.first()
+            }
+            delay(2.seconds)
+            while (player.playbackEngine.assetPosition < 2) {
+                delay(10.milliseconds)
+            }
+            player.playbackEngine.pause()
+            delay(1.seconds)
+            player.playbackEngine.play()
+            withTimeout(8000) {
```

Comment:
Logaf low: We have already played 2 seconds out of the 5 second track, so this could in theory be lowered to 6000, as opposed to the previous test where it plays the whole thing. I'm fine with keeping it at 8000 as well.

Reply:
> Logaf low: We have already played 2 seconds out of the 5 second track, so this could in theory be lowered to 6000, as opposed to the previous test where it plays the whole thing. 

Tricky topic. Those 2 seconds as per the track might've taken more, if the delay leans more on the start of playback for example, so it would theoretically not be correct to just substract 2 seconds from the timeout. I would say we don't have an exact science so no number is correct really, so this one works for me out of why not basically

> Also, I see you are using x.seconds in other places, should we do that for these as well?

will do

Reply:
done

Reply:
I agree, let's keep the numbers as is.